pkg/controller/controller.go

```diff
@@ -383,12 +455,21 @@ func (c *Controller) getDesiredWorkers(
 		}
 
 		desiredWorkers := int32(math.Ceil(usageRatio * float64(currentWorkers)))
-		if !disruptable && desiredWorkers < currentWorkers {
-			return currentWorkers
+		if desiredWorkers < minAvailableWorkers {
```

Comment:
Instead of `minAvailableWorkers` representing another absolute min value that the workers can partially scale down to, it can be a value(`maxDisruption` or `partialScaledownDiff`) that specifies how much partial scale down is allowed(either in absolute number of pods or as a percentage of current pods) at a time(per cycle of the control loop). If `maxDisruption` is 0, it means the pods will not do partial scaledown and if the `maxDisruption` is 10%, partial scaledown can at most remove 10% of currently running pods at a time. If `maxDisruption` is unspecified, it can default to 0(non-disruptable by default) or 100%(disruptable by default) using a global `defaultMaxDisruption` command-line flag.

Reply:
fixed in https://github.com/practo/k8s-worker-pod-autoscaler/pull/62/commits/47987f9711e13285d32c48b6d7a681addaa98584